student.rb

```diff
@@ -1,12 +1,19 @@
 require_relative 'person'
 
 class Student < Person
-  def initialize(id, age, classroom, name = 'Unknown', parent_permission: true)
+  attr_reader :classroom
+
+  def initialize(id, age, classroom = nil, name = 'Unknown', parent_permission: true)
     @classroom = classroom
+    add_student_classroom if clasroom
     super(id, age, name, parent_permission)
   end
 
   def play_hooky
     '¯(ツ)/¯'
   end
+
+  def add_student_classroom(clasroom)
+    clasroom.add_student(self) unless classroom.students.include?(self)
+  end
```

Comment:
I suggest we add an instance of the classroom variable here, since we want to access it in this method😉.

```suggestion
  def add_student_classroom(classroom)
    @classroom = classroom
    clasroom.add_student(self) unless classroom.students.include?(self)
  end
```